### PR TITLE
2.4.7 merge

### PR DESCRIPTION
**Pull request by JamieRuderman**
Keep user signed in on desktop but not from browser
Fix uninstalling bin directory on Mac/Linux

### CODE REVIEW

File: frontend/src/store.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import logger from 'redux-logger'
 const plugins = [immerPlugin(), selectPlugin()]
 let middlewares: any[] = []
 
-if (process.env.NODE_ENV !== 'test') middlewares.push(logger)
+// if (process.env.NODE_ENV !== 'test') middlewares.push(logger)
```
Comment (danawoodman) [inline frontend/src/store.tsx]: Intentional?

File: frontend/src/models/auth.ts [hunk 2]
```diff
@@ -72,7 +96,10 @@ export default createModel({
     async signedIn() {
       dispatch.auth.signInFinished()
       dispatch.auth.checkSignInFinished()
-      dispatch.devices.shouldSearchDevices()
+      if (!r3.token) await dispatch.auth.fetchAuthToken()
+      const searchOnly = await dispatch.devices.shouldSearchDevices()
+      console.log('SEARCH ONLY?', searchOnly)
```
Comment (danawoodman) [inline frontend/src/models/auth.ts]: Remove log

File: frontend/src/models/devices.ts [hunk 3]
```diff
@@ -44,16 +44,13 @@ export default createModel({
...
-        dispatch.devices.fetch()
-        return
-      }
+      let searchOnly = pref !== 'false'
 
       // If they have too many services, show search only.
-      return r3.devices
-        .count()
-        .then(count => dispatch.devices.setSearchOnly(count.services > SEARCH_ONLY_SERVICE_LIMIT))
+      await r3.devices.count().then(count => (searchOnly = count.services > SEARCH_ONLY_SERVICE_LIMIT))
```
Comment (danawoodman) [inline frontend/src/models/devices.ts]: Why not use await, assign count to var and then do your variable setting? Seems more idiomatic async/await

File: frontend/src/store.tsx (hunk 1 above)
Comment (JamieRuderman) [reply frontend/src/store.tsx]: No didn't mean to commit - backed it out already. Thanks!

File: frontend/src/models/devices.ts (hunk 3 above)
Comment (JamieRuderman) [reply frontend/src/models/devices.ts]: Fixed!

**Pull request merged**